tests/codegen/align-byval.rs

```diff
@@ -0,0 +1,45 @@
+// revisions:m68k x86_64-linux x86_64-windows
+
+//[m68k] compile-flags: --target m68k-unknown-linux-gnu
+//[m68k] needs-llvm-components: x86
+//[x86_64-linux] compile-flags: --target x86_64-unknown-linux-gnu
+//[x86_64-linux] needs-llvm-components: x86
+//[x86_64-windows] compile-flags: --target x86_64-pc-windows-msvc
+//[x86_64-windows] needs-llvm-components: x86
+
+// Tests that `byval` alignment is properly specified (#80127).
+// The only targets that use `byval` are m68k, x86-64, and x86. Note that
+// x86 has special rules (see #103830), and it's therefore ignored here.
+// Note also that Windows mandates a by-ref ABI here, so it does not use byval.
```

Comment:
Note: I removed wasm here because it does not use byval in this situation, the codegen is

```ll
; ModuleID = 'align_byval.9946dc3cac1a96f8-cgu.0'
source_filename = "align_byval.9946dc3cac1a96f8-cgu.0"
target datalayout = "e-m:e-p:32:32-p10:8:8-p20:8:8-i64:64-n32:64-S128-ni:1:10:20"
target triple = "wasm32-unknown-unknown"

%Foo = type { [16 x i32] }

; align_byval::main
; Function Attrs: nounwind
define hidden void @_ZN11align_byval4main17h05226bf3749dbc8fE() unnamed_addr #0 {
start:
  tail call void @f(%Foo { [16 x i32] [i32 1, i32 1, i32 1, i32 1, i32 1, i32 1, i32 1, i32 1, i32 1, i32 1, i32 1, i32 1, i32 1, i32 1, i32 1, i32 1] }) #1
  ret void
}

; Function Attrs: nounwind
declare dso_local void @f(%Foo) unnamed_addr #0

attributes #0 = { nounwind "target-cpu"="generic" }
attributes #1 = { nounwind }
```

Reply:
Maybe adjust the struct so it's no longer a homogeneous aggregate?

Reply:
Ah, it's the wasm32-unknown-unknown target that basically disables the C ABI by setting `options.default_adjusted_cabi = Some(Abi::Wasm);` Changed to emscripten and added it back.